### DISCUSSION TITLE
How to use rad_max from an IRF, v0.20.1 ?

### DISCUSSION THREAD

**Discussion by mdpunch** [gammapy/gammapy#4059] [Help] [answered]
Hello,
I'm testing out gammapy v 0.20.1, in particular trying the rad_max IRF for a variable rad_max (theta cut) as a function of reconstructed energy.
I have my IRFs, including one with RadMax2D.
```
print(irfs)

{'aeff': <gammapy.irf.effective_area.EffectiveAreaTable2D object at 0x7f3e56ff2e50>, 
'edisp': <gammapy.irf.edisp.core.EnergyDispersion2D object at 0x7f3e57003b80>, 
'psf': <gammapy.irf.psf.table.PSF3D object at 0x7f3e56f7d340>, 
'rad_max': <gammapy.irf.rad_max.RadMax2D object at 0x7f3e56fcb7c0>, 
'bkg': <gammapy.irf.background.Background2D object at 0x7f3e56f6ff70>}
```
and `irfs['rad_max'].plot_rad_max_vs_energy()` gives the expected result.
But, when I try to create an observation `obs = Observation.create(pointing=pointing, irfs=irfs, livetime=livetime)` I can access all the IRFs (e.g., `obs.psf` or `obs.aeff`) except for the rad_max:
```
obs.rad_max

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/tmp/ipykernel_780266/1763588394.py in <cell line: 2>()
      1 #Tk
----> 2 obs.rad_max

/scr/punch/miniconda3/envs/alto/lib/python3.9/site-packages/gammapy/data/observations.py in rad_max(self)
    103         edisp = self.edisp
    104         if edisp is not None and edisp.is_pointlike:
--> 105             self._rad_max = RadMax2D.from_irf(self.edisp)
    106 
    107         return self._rad_max

/scr/punch/miniconda3/envs/alto/lib/python3.9/site-packages/gammapy/irf/rad_max.py in from_irf(cls, irf)
     63 
     64         if "RAD_MAX" not in irf.meta:
---> 65             raise ValueError("Irf does not contain RAD_MAX keyword")
     66 
     67         rad_max_value = irf.meta["RAD_MAX"]

ValueError: Irf does not contain RAD_MAX keyword
```
..,. and the rad_max IRF isn't in the list of the ones available:
```
obs.available_irfs

['aeff', 'edisp', 'psf', 'bkg']
```
Also, the background estimation later on doesn't use this information, and I have to calculate the counts_off myself for the dataset.
What am I missing?
Thanks in advance for help,
  Michael.

**Comment by bkhelifi** [comment, marked as answer]
Hi @mdpunch,
Your code is using the function `observation.create` to add your custom IRFs into your 'simulated' observation. Unfortunately, the rad_max irf is added inside the create function. This is a feature to add...

**Reply by mdpunch** [reply]
Thanks for the answer, and the resulting issue: https://github.com/gammapy/gammapy/issues/4060

As we discussed and agreed, having the `RAD_MAX` as in IRF may be problematic, since it is really metadata going into the creation of `aeff` and `edisp`.  It seems to be possible to add there too.  So if it is in either/or/both as metadata / IRF, it needs to be handled both ways, I guess.

But that's maybe a philosophical point, whereas it might be best to go with "whatever works", which the issue will solve.
